### PR TITLE
fix for stack index offset

### PR DESCRIPTION
**Pull request by ccunniff**

### CODE REVIEW

File: logger/src/main/java/com/orhanobut/logger/Logger.java [hunk 1]
```diff
@@ -120,7 +120,7 @@ public static void e(String tag, Exception e) {
 
     public static void e(String message, int methodCount) {
         validateMethodCount(methodCount);
-        e(message, null, methodCount);
+        e(TAG, message, methodCount);
```
Comment (orhanobut) [inline logger/src/main/java/com/orhanobut/logger/Logger.java]: good catch, totally missed this one.

File: logger/src/main/java/com/orhanobut/logger/Logger.java [hunk 2]
```diff
@@ -303,8 +303,21 @@ private static void logHeaderContent(int logType, String tag, int methodCount) {
             logDivider(logType, tag);
         }
         String level = "";
+
+        int stackOffset = -1;
+        boolean foundThisClass = false;
```
Comment (orhanobut) [inline logger/src/main/java/com/orhanobut/logger/Logger.java]: would be better to create another method in order to determine stack offset. and I would also suggest another approach to find the correct stack index.

We can iterate by starting from 3 (there is always 2 stack and sometimes another stack comes from somewhere that I couldn't find the real reason). We guarantees that starting trace will be a logger method. Then we simplify the method.
```java
    private static int getStackIndex(StackTraceElement[] trace) {
        for (int i = 3; i < trace.length; i++) {
            StackTraceElement e = trace[i];
            if (!e.getClassName().equals(Logger.class.getName())) {
                return --i;
            }
        }
        return -1;
    }
```

**Comment by ccunniff** [comment]
I like your implementation of
```java
private static int getStackOffset(StackTraceElement[] trace) {
        for (int i = MIN_STACK_OFFSET; i < trace.length; i++) {
            StackTraceElement e = trace[i];
            if (!e.getClassName().equals(Logger.class.getName())) {
                return --i;
            }
        }
        return -1;
    }
```

**Comment by orhanobut** [comment]
:+1:

**Pull request merged**